src/pages/workspace/accounting/netsuite/import/NetSuiteImportCustomFieldPage.tsx

```diff
@@ -0,0 +1,146 @@
+                    />
+                }
+                containerStyle={[styles.flex1, styles.justifyContentCenter]}
+            />
+        ),
+        [importCustomField, styles, translate],
+    );
+
+    const listHeaderComponent = useMemo(
+        () => (
+            <View style={[styles.ph5, styles.flexRow]}>
+                <HelpLinkComponent
+                    importCustomField={importCustomField}
+                    styles={styles}
+                    translate={translate}
+                    alignmentStyle={styles.textAlignLeft}
+                />
+            </View>
+        ),
+        [styles, importCustomField, translate],
+    );
+
+    return (
+        <ConnectionLayout
+            displayName={NetSuiteImportCustomFieldPage.displayName}
+            headerTitle={`workspace.netsuite.import.importCustomFields.${importCustomField}.title`}
+            accessVariants={[CONST.POLICY.ACCESS_VARIANTS.ADMIN, CONST.POLICY.ACCESS_VARIANTS.PAID]}
+            policyID={policyID}
+            featureName={CONST.POLICY.MORE_FEATURES.ARE_CONNECTIONS_ENABLED}
+            contentContainerStyle={[styles.flex1]}
+            titleStyle={styles.ph5}
+            connectionName={CONST.POLICY.CONNECTIONS.NAME.NETSUITE}
+            shouldIncludeSafeAreaPaddingBottom
+            onBackButtonPress={() => Navigation.goBack(ROUTES.POLICY_ACCOUNTING_NETSUITE_IMPORT.getRoute(policyID))}
+        >
+            {data.length === 0 ? listEmptyComponent : listHeaderComponent}
+
+            {data.map((record, index) => (
+                <MenuItemWithTopDescription
+                    // eslint-disable-next-line react/no-array-index-key
+                    key={`${record.internalID}-${index}`}
+                    description={translate(`workspace.netsuite.import.importCustomFields.${importCustomField}.recordTitle`)}
+                    shouldShowRightIcon
+                    title={'listName' in record ? record.listName : record.segmentName}
+                    onPress={() => Navigation.navigate(ROUTES.POLICY_ACCOUNTING_NETSUITE_IMPORT_CUSTOM_FIELD_VIEW.getRoute(policyID, importCustomField, index))}
+                />
+            ))}
+
+            <FixedFooter style={[styles.mtAuto, styles.pt3]}>
+                <Button
+                    success
+                    large
+                    onPress={() => {
+                        if (importCustomField === CONST.NETSUITE_CONFIG.IMPORT_CUSTOM_FIELDS.CUSTOM_SEGMENTS) {
+                            Navigation.navigate(ROUTES.POLICY_ACCOUNTING_NETSUITE_IMPORT_CUSTOM_SEGMENT_ADD.getRoute(policyID));
+                        } else {
+                            Navigation.navigate(ROUTES.POLICY_ACCOUNTING_NETSUITE_IMPORT_CUSTOM_LIST_ADD.getRoute(policyID));
+                        }
+                    }}
+                    text={translate(`workspace.netsuite.import.importCustomFields.${importCustomField}.addText`)}
+                />
+            </FixedFooter>
```

Comment:
I think we shouldn't disable the buttons. The requests are getting queued. Anyway we should be looking at checking how to process arrays with pendingFields instead of disabling. wdyt?